src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1264,28 +1283,139 @@ public static bool IsPatternContext(this SyntaxTree syntaxTree, SyntaxToken left
                 return true;
             }
 
+            // case ($
+            if (leftToken.IsKind(SyntaxKind.OpenParenToken) && node.IsParentKind(SyntaxKind.CaseSwitchLabel))
+            {
+                return true;
+            }
+
             // e switch { $$
             // e switch { ..., $$
-            if (leftToken.IsKind(SyntaxKind.OpenBraceToken, SyntaxKind.CommaToken) && leftToken.Parent.IsKind(SyntaxKind.SwitchExpression))
+            if (leftToken.IsKind(SyntaxKind.OpenBraceToken, SyntaxKind.CommaToken) && node.IsKind(SyntaxKind.SwitchExpression))
             {
                 return true;
             }
 
             // e is ($$
             // e is (..., $$
-            if (leftToken.IsKind(SyntaxKind.OpenParenToken, SyntaxKind.CommaToken) && leftToken.Parent.IsKind(SyntaxKind.PositionalPatternClause))
+            if (leftToken.IsKind(SyntaxKind.OpenParenToken, SyntaxKind.CommaToken) && node.IsKind(SyntaxKind.PositionalPatternClause))
             {
                 return true;
             }
 
             // e is { P: $$
             // e is { ..., P: $$
-            if (leftToken.IsKind(SyntaxKind.ColonToken) && leftToken.Parent.IsKind(SyntaxKind.NameColon) &&
-                leftToken.Parent.IsParentKind(SyntaxKind.Subpattern))
+            if (leftToken.IsKind(SyntaxKind.ColonToken) && node.IsKind(SyntaxKind.NameColon) &&
+                node.IsParentKind(SyntaxKind.Subpattern))
+            {
+                return true;
+            }
+
+            // e is { P: ($$
+            if (leftToken.IsKind(SyntaxKind.OpenParenToken) && node.IsParentKind(SyntaxKind.ConstantPattern))
+            {
+                return true;
+            }
+
+#if !CODE_STYLE
+            // if (e is { P: (1 or $$
+            if ((leftToken.IsKind(SyntaxKind.AndKeyword) || leftToken.IsKind(SyntaxKind.OrKeyword)) &&
+                node is BinaryPatternSyntax)
             {
                 return true;
             }
 
+            if (node.IsParentKind(SyntaxKind.IsPatternExpression))
+            {
+                // e is 1 and $$
+                if (node.IsKind(SyntaxKind.AndPattern))
```

Comment:
1. i don't think think this needs the `IsPatternExpression` check above.  
2. you already checked for this case above.